azurerm/resource_arm_mysql_server.go

```diff
@@ -42,12 +41,24 @@ func resourceArmMySqlServer() *schema.Resource {
 							Type:     schema.TypeString,
 							Required: true,
 							ValidateFunc: validation.StringInSlice([]string{
-								"MYSQLB50",
-								"MYSQLB100",
-								"MYSQLS100",
-								"MYSQLS200",
-								"MYSQLS400",
-								"MYSQLS800",
+								"B_Gen4_1",
+								"B_Gen4_2",
+								"B_Gen5_1",
+								"B_Gen5_2",
+								"GP_Gen4_2",
+								"GP_Gen4_4",
+								"GP_Gen4_8",
+								"GP_Gen4_16",
+								"GP_Gen4_32",
+								"GP_Gen5_2",
+								"GP_Gen5_4",
+								"GP_Gen5_8",
+								"GP_Gen5_16",
+								"GP_Gen5_32",
+								"MO_Gen5_2",
+								"MO_Gen5_4",
+								"MO_Gen5_8",
+								"MO_Gen5_16",
```

Comment:
Right, but there specific configuration that only supported by certain tiers, take Memory Optimized(MO) for example... MO tiers can only be Gen5 family with between 2 and 16 vCores, with the regex it would allow for invalid configurations which would fail when apply was ran. If that is acceptable I would be happy to switch over to the regex, however I was under the impression that it was not so I went this route and forced only valid combinations to be entered into the schema. Let me know what you think.